### PR TITLE
Added cuda auto-installation

### PR DESCRIPTION
**Pull request by rsnk96**
Addresses #75 

Auto installs the latest cuda

### CODE REVIEW

File: 3-ML-Basic.sh [hunk 1]
```diff
@@ -31,6 +31,22 @@ run_and_echo () {
 }
 
 
+if [[ (-n $(lspci | grep -i nvidia)) && (! ( -d "/usr/local/cuda" ) ) ]]; then
+    echo "Installing the latest cuda"
+    cuda_instr_block=$(wget -q -O - 'https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64&target_distro=Ubuntu&target_version=1804&target_type=runfilelocal' | grep wget | head -n 1)
```
Comment (rajat2004) [inline 3-ML-Basic.sh]: Should it be different for `16.04` and `18.04`?

**Review by rajat2004 — APPROVED** [review]
Needs rebase

**Comment by rajat2004** [comment]
Definitely the most complex `sed` usage I've seen so far! Can't even start to understand how it's all working, just that it is

File: 3-ML-Basic.sh (hunk 1 above)
Comment (rajat2004) [reply 3-ML-Basic.sh]: Well, it's actually the same for both, so doesn't matter I guess

**Comment by rsnk96** [comment]
> Definitely the most complex `sed` usage I've seen so far! Can't even start to understand how it's all working, just that it is

Just followed [this](https://stackoverflow.com/a/27642076) SO answer :)

File: 3-ML-Basic.sh (hunk 1 above)
Comment (rsnk96) [reply 3-ML-Basic.sh]: Yeah...but the URL can be made version agnostic nonetheless, good suggestion

**Pull request merged**